### PR TITLE
Add precision for create_dir function

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Took me a while to find the other equivalent so it seems to be necessary.

r? @QuietMisdreavus

### CODE REVIEW

File: src/libstd/fs.rs [hunk 1]
```diff
@@ -1755,6 +1755,10 @@ pub fn canonicalize<P: AsRef<Path>>(path: P) -> io::Result<PathBuf> {
 ///
 /// [changes]: ../io/index.html#platform-specific-behavior
 ///
+/// **NOTE**: If an element in the given path doesn't exist, the function will
+/// fail. To create a path and all its non-existing parent, take a look at the
+/// [`create_dir_all`] function.
```
Comment (QuietMisdreavus) [inline src/libstd/fs.rs]: Language polish...

```markdown
**NOTE**: If a parent of the given path doesn't exist, this function will return an error. To create a directory and all its missing parents at the same time, use the [`create_dir_all`] function.
```

**Comment by GuillaumeGomez** [comment]
Thanks for the polishing @QuietMisdreavus! Updated.

**Comment by QuietMisdreavus** [comment]
Oh hey, looking again, there's a perfectly usable "Errors" section that already exists, right below the note you added! How about adding this to that list instead?

```markdown
- A parent directory does not exist. To create a directory and all its parent directories at the same time, use [`create_dir_all`].
```

**Comment by GuillaumeGomez** [comment]
Ah sure. The problem is that it looks exactly like the `create_dir_all` docs. I wanted to make it stand out. I can add both if you [refer?

**Comment by QuietMisdreavus** [comment]
"Looking the same as the other docs" is the point. If we have the note *somewhere*, people will be able to find it. I don't want to have two different sections that describe when a function returns an error; the one you're adding is just redundant. We have a perfectly serviceable documentation standard when describing the situations where a function returns an error, please use it.

**Comment by GuillaumeGomez** [comment]
Added it to the error list.

**Comment by QuietMisdreavus** [comment]
The other note is still there, but looking at it again, i think it's probably fine to leave it. If it seems redundant to enough other people, we can take it out later. Thanks!

@bors r+ rollup

**Pull request merged**